### ISSUE TITLE
Move to Jakarta Faces (v4+) / Remove JSF (javax)

### ISSUE THREAD

**Issue by tandraschko** [primefaces/primefaces#11915]
As currently discussed in https://github.com/primefaces/primefaces/issues/11877:
When should we remove Javax support? for 16.0? 17.0?

It seems that the usage is quite low actually:
https://groups.google.com/g/jakartaee-ambassadors/c/-1cU-XkT810?pli=1

- [X] migrate to jakarta packages in code
- [x] migrate taglibs, xhtmls, web.xml decleration & namespaces
- [ ] ~~remove p:resetInput in favor of f:resetValues~~ -> We can't as f:resetValues doesnt have clearModel yet
- [X] remove HTML5_COMPLIANCE 
- [ ] can we remove COOKIES_SAME_SITE in favor of getting the config for the sessionId?
- [ ] ~~Remove legacy primefaces XHTML namespace?~~ -> I woudlnt do it for now, JSF also still supports the oldschool namespaces

**Comment by melloware** [comment]
I think until that usage increases to a significant level we keep support of `javax`.

**Comment by tandraschko** [comment]
Currently only 7% of the users are on javax

**Comment by melloware** [comment]
that seems REALLY low.  I know NONE of my clients are on Jakarta yet.  So those results may be skewed?

**Comment by tandraschko** [comment]
maybe
i think that the most people that voted are propably the users, which are interessted in new jakarta stuff anyway

**Comment by christophs78** [comment]
"my" public sector organisation is currently moving to spring boot 3.x. (from 2.x)
spring boot 2.x (=javax) is EOL since ~11/2023. (there is still some commerical support for 2.x)
But i know the combination of JSF with spring (boot) is unusal.

Tomcat 9 will propably be there for a few more years. (2027? they just set 8.5 to EOL)
Open Liberty maybe the same.

Some other public sector organisations i know already moved vom JavaEE to JakartaEE some time ago.

Some other maybe plan to use there legaxy JavaEE / JSF applications as long as viable and do then something new maybe based on JS-frontends. Maybe reuse some of the backend-stuff if refactoring this to REST / GraphQL / ... is viable.

Waiting until 2027+ to phase out javax on our side IMO is to far away.

"Things" like Hibernate 5 (last javax-verion) get very limited support. Similar things for other core parts of JavaEE. Don´t know how  much effort Redhat, IBM & Co still invest to give JBoss, Open Liberty, still limited life-support. Or they hope there are not much CVE´s and Co they need to adresse.

**Comment by tandraschko** [comment]
TomEE 8.0.x (javax) is or will be EOL soon, too

i think we could target 2025 to remove it

**Comment by melloware** [comment]
That sounds like a good plan to me.

**Comment by Bukama** [comment]
7% usage of `javax` seems very low to me, but I can't proof it either right or wrong.
You could also argue that you drop it now as PF 14 needs Java 11 which does not brings `javax` build in.

**Comment by escay** [comment]
People on Weblogic 14.1.1.0.0 running Java 11 are stuck on using javax / JEE8. They need to migrate to an alternative application server. Weblogic 14.1.1.0.0 end of life is tier 1 support is March 2025 / tier 2 support is planned until 2028 (https://docs.oracle.com/cd/F52330_01/reference_guides/insbridge_hdwr_sfwr_guide/Content/Guides_Sales/Insbridge%20Hardware%20Software%20Specs/End%20of%20Support%20Schedule.htm).
And no Weblogic 15 is planned, support issue https://support.oracle.com/knowledge/Middleware/2935815_1.html states: _"There is no specific plan for JakartaEE9 or 10"_ or it is way too late and people migrated to alternatives.

Here a target of 2025 seems also reasonable.

TomEE people can upgrade to TomEE (Jakarta 10) or newer and do not need javax anymore.
Payara people can upgrade to Payara 6 (Jakarta 10) or newer and do not need javax anymore.
Glassfish people can upgrade to Glassfish 7 (Jakarta 10) or newer and do not need javax anymore.
Same for Open Liberty and others: there are Jakarta 10 upgrade options available for over 2 years already.

I do not know what the overhead is for supporting javax. But thank you for letting me still using latest PF in Weblogic.

**Comment by tandraschko** [comment]
:/ i would really suggest to migrate to some other AS then

**Comment by lprimak** [comment]
ASAP. Old users won't upgrade anyway so it won't be an issue

**Comment by jepsar** [comment]
We can also consider to create a branch for javax in test and let the master use Jakarta and drop its branch.

https://github.com/primefaces/primefaces-test/tree/jakarta

**Comment by tandraschko** [comment]
yeah plz, showcase is already jakarta

**Comment by melloware** [comment]
OK i will switch them i will have PF Test master be the Jakarta version and the Javax version be in a branch.

**Comment by jepsar** [comment]
And maybe convert the other branches (like JPA) to Jakarta as well.

**Comment by melloware** [comment]
OK `master` is now Jakarta branch. a Javax branch is created and i deleted the old Jakarta branch.  I will update the JPA to Jakarta as well.

**Comment by melloware** [comment]
JPA converted to Jakarta now.

**Comment by christophs78** [comment]
Side note: Debugging from a Jakarta-application into shaded PF-code feels awkward. Looking into this i would appreciate to make the move to Jakarta as default with PF 16.

**Comment by melloware** [comment]
Yeah I get burned by this all the time trying to debug the new Showcase because my IDE is confused about the code being Jakarta but all the code in the IDE is javax.

**Comment by tandraschko** [comment]
just to be sure
we will remove javax support completely
and also merge the javax and jakarta integration tests

**Comment by tandraschko** [comment]
the mininum version will be Faces 4.0
we should skip 3.0, right?

**Comment by melloware** [comment]
Yes faces 4.0 should be the minimum and one set of integration tests.  

I can help with the integration tests by open rewrite script should do 99% of the conversion?
